Branch-SDK/src/main/java/io/branch/referral/ServerRequest.java

```diff
@@ -774,11 +799,13 @@ public void onPreExecute() {
                 String externalIntentUri = prefHelper_.getExternalIntentUri();
                 utility.parseReferringURL(externalIntentUri);
 
-                JSONObject urlQueryParams = utility.getURLQueryParamsForRequest(this);
+                if (prefHelper_.getConsumerProtectionAttributionLevel() == Defines.BranchAttributionLevel.FULL || !prefHelper_.isAttributionLevelInitialized()) {
```

Comment:
Is it possible for `getConsumerProtectionAttributionLevel` to have a value and `isAttributionLevelInitialized` not be true?

In other words, if `prefHelper_.getConsumerProtectionAttributionLevel() == Defines.BranchAttributionLevel.FULL` how can `prefHelper_.isAttributionLevelInitialized()` be false? Correct me if I'm wrong but it seems redundant

Reply:
Basically we want this if statement to be true if the CPP level is FULL or if it hasn't been set, since that's treated like FULL. However, we don't want to always add the `cpp_level` field to requests, so i didn't just set the default for `getConsumerProtectionAttributionLevel()` to be FULL. 

So thats why im checking for either the CPP level to be full OR for it to not be set here.